src/hotspot/share/cds/filemap.cpp

```diff
@@ -1012,134 +1039,195 @@ void FileMapInfo::validate_non_existent_class_paths() {
   }
 }
 
+// a utility class for checking file header
+class FileHeaderHelper {
+  int _fd;
+  GenericCDSFileMapHeader* _header;
```

Comment:
This should be changed to `GenericCDSFileMapHeader _header;`. That way, you don't need to malloc and free it.